src/main/java/org/folio/rest/impl/InvoicesImpl.java

```diff
@@ -81,13 +87,19 @@ public void putInvoiceInvoicesById(String id, String lang, Invoice invoice, Map<
     InvoiceHelper invoiceHelper = new InvoiceHelper(okapiHeaders, vertxContext, lang);
 
     // Validate incoming invoice first to avoid extra calls to other services if content is invalid
-    if (!invoiceHelper.validateIncomingInvoice(invoice)) {
-      asyncResultHandler.handle(succeededFuture(invoiceHelper.buildErrorResponse(422)));
-      return;
-    }
-
-    invoiceHelper.updateInvoice(invoice)
-      .thenAccept(ok -> asyncResultHandler.handle(succeededFuture(invoiceHelper.buildNoContentResponse())))
+    invoiceHelper.validateIncomingInvoice(invoice)
```

Comment:
why is this change needed?

Reply:
Once `validateIncomingInvoice` return type changed to `CompletableFuture<Boolean>`, it starting erroring out with `The operator ! is undefined for the argument type(s) CompletableFuture<Boolean>`. So to fix it, I had to refactor to use CompletableFuture.